app/components/Form.js

```diff
@@ -0,0 +1,237 @@
+            } else {
+                this.setState({ emailValidation: false });
+            }
+        }else if(type=="password"){
+            if (num.test(text)) {
+                this.setState({ passwordValidate: true, password: text,showError: false});
+            }else{
+                if (this.props.type == 'Signup') {
+                    let msg = 'password must contain at least: 1 uppercase character, 1 number ,must be 8 characters or longer';
+                    this.showErrorMessage(msg, true);
+                }
+                this.setState({ passwordValidate: false });
+            }
+        }else if(type == "name"){
+            this.setState({ name : text})
+        }
+    }
+
+    async verifyUser(key){
+        try {
+            await AsyncStorage.getItem(key).then((val) => {
+                this.setState({ user: JSON.parse(val)});
+                });
+        } catch (error) {
+            console.log('Error: ', error);
+        }
+        
+    }
+
+    
+    async saveUser (data,key) {
+        try {
+            let user = JSON.stringify(data);
+            
+            await AsyncStorage.setItem(key, user);
+        } catch (error) {
+            // Error saving data
+            console.log(error);
+            
+        }
+    };
+
+    showErrorMessage(msg,state){
+        this.setState({
+            errorMsg: msg,
+            showError: state
+        });
+    }
+
+    saveLogState(){        
+        this.props.screenProps.isLoggedIn();
+    }
+
+    buttonHandler(){
+
+        let {type} = this.props;
+        let { name, email, password } = this.state;
+        
+        if (type =='Login'){
+
+            if (email != '') {
+                if (password != '') {
+
+                    this.verifyUser(email).then(() => {
+                        let { user: user } = this.state;
+                        if (user != null) {
+
+                            if ((user.email == email) && (user.password == password)) {
+                                this.setState({ showError: false });
+                                this.saveLogState();
+                                this.props.navigation.navigate('Home');
+                            } else {
+                                let msg = `We couldn’t find an account matching the email and password you entered. Please verify and try again.`;
+                                this.showErrorMessage(msg, true)
+                            }
+
+
+                        } else {
+
+                            let msg = `We couldn’t find an account matching the email and password you entered. Please verify and try again.`;
+                            this.showErrorMessage(msg, true)
+                        }
+                    });  
+
+                } else {
+                    let msg = `All fields are required, please verify and try again.`;
+                    this.showErrorMessage(msg, true)
+                }
+            } else {
```

Comment:
you have a lot of else, this is not a good practice, instead you could use a object key value with the messages or a switch case.